### PR TITLE
Add TE fuser ops

### PR DESCRIPTION
**Pull request by eellison**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #68826
* __->__ #68825

Factoring out the elementwise ops in tensorexpr fuser and adding their corresponding shape functions, since we need shape functions to fuse them with dynamic shapes

Differential Revision: [D32732466](https://our.internmc.facebook.com/intern/diff/D32732466)

### CODE REVIEW

**Comment by ZolotukhinM** [comment]
Do you mind adding more details to the commit message?

File: torch/csrc/jit/runtime/symbolic_shape_registry_util.cpp [hunk 1]
```diff
@@ -0,0 +1,124 @@
+
+
```
Comment (ZolotukhinM) [inline torch/csrc/jit/runtime/symbolic_shape_registry_util.cpp:2]: delet dis

File: torch/csrc/jit/runtime/symbolic_shape_registry_util.cpp [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+#include <unordered_map>
+
+namespace torch {
+namespace jit {
+
+const OperatorMap<std::string>& get_tensorexpr_elementwise_set() {
+  // clang-format off
+ static const OperatorMap<std::string> tensorexpr_elementwise_set{
+     {"aten::add.Scalar(Tensor self, Scalar other, Scalar alpha=1) -> Tensor", "unary"},
+      {"aten::_cast_Float(Tensor self, bool non_blocking) -> Tensor", "unary"},
```
Comment (ZolotukhinM) [inline torch/csrc/jit/runtime/symbolic_shape_registry_util.cpp:17]: Nit: formatting seems 1 char off here.

File: torch/csrc/jit/runtime/symbolic_shape_registry_util.cpp [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+      {"aten::__rshift__.Tensor(Tensor self, Tensor other) -> Tensor", "broadcast"},
+      // TODO: enable other min/max variants, operators that can be both
+      // elementwise or reductions:
+      {"aten::min.other(Tensor self, Tensor other) -> Tensor", "broadcast"},
+      {"aten::max.other(Tensor self, Tensor other) -> Tensor", "broadcast"},
+      {"aten::lerp.Tensor(Tensor self, Tensor end, Tensor weight) -> Tensor", "broadcast_three"},
+      {"aten::addcmul(Tensor self, Tensor tensor1, Tensor tensor2, *, Scalar value=1) -> Tensor", "broadcast_three"},
+      {"aten::where.self(Tensor condition, Tensor self, Tensor other) -> Tensor", "broadcast_three"},
+      {"aten::where.ScalarSelf(Tensor condition, Scalar self, Tensor other) -> Tensor", "broadcast_one_three"},
+      // TODO: enable slice, shape inference is not implemented for this op yet
```
Comment (ZolotukhinM) [inline torch/csrc/jit/runtime/symbolic_shape_registry_util.cpp:118]: The list didn't change during the move, right?

**Review by ZolotukhinM — APPROVED** [review]
Looks good!

File: torch/csrc/jit/runtime/symbolic_shape_registry_util.cpp (hunk 3 above)
Comment (eellison) [reply torch/csrc/jit/runtime/symbolic_shape_registry_util.cpp:118]: yes - just copy-pasting

Comment (eellison) [reply torch/csrc/jit/runtime/symbolic_shape_registry_util.cpp:117]: no, ctrl-F "aten::" 98 results in both lists

**Comment by eellison** [comment]
@eellison has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32732466).

**Pull request closed without merging**